spec/unit/pops/evaluator/evaluating_parser_spec.rb

```diff
@@ -1040,6 +1040,12 @@ def assert_no_undef(x)
       expect{parser.evaluate_string(scope, "assert_no_undef({1 => undef})")}.to_not raise_error()
     end
 
+    it 'a lambda return value is checked using the return type' do
+      expect(parser.evaluate_string(scope, "[1,2].map |Integer $x| >> Integer { $x }")).to eql([1,2])
+      expect { parser.evaluate_string(scope, "[1,2].map |Integer $x| >> String { $x }") }.to raise_error(
+        /value returned from lambda had wrong type, expected a String value, got Integer/)
+    end
+
```

Comment:
example that when return is not specified makes the return type default to Any is missing (there should be an explicit test even if all other examples do not have return type specified).